packages/workflows/src/definition/price-list/create-price-lists.ts

```diff
@@ -0,0 +1,78 @@
+import { PricingTypes, WorkflowTypes } from "@medusajs/types"
+import {
+  TransactionStepsDefinition,
+  WorkflowManager,
+} from "@medusajs/orchestration"
+import { exportWorkflow, pipe } from "../../helper"
+
+import { PriceListHandlers } from "../../handlers"
+import { Workflows } from "../../definitions"
+
+export enum CreatePriceListActions {
+  prepare = "prepare",
+  createPriceList = "createPriceList",
+}
+
+const workflowSteps: TransactionStepsDefinition = {
+  next: {
+    action: CreatePriceListActions.prepare,
+    noCompensation: true,
+    next: {
+      action: CreatePriceListActions.createPriceList,
+    },
+  },
+}
+
+const handlers = new Map([
+  [
+    CreatePriceListActions.prepare,
+    {
+      invoke: pipe(
+        {
+          inputAlias: CreatePriceListActions.prepare,
+          merge: true,
+          invoke: {
+            from: CreatePriceListActions.prepare,
+          },
+        },
+        PriceListHandlers.prepareCreatePriceLists
+      ),
+    },
+  ],
+  [
+    CreatePriceListActions.createPriceList,
+    {
+      invoke: pipe(
+        {
+          invoke: {
+            from: CreatePriceListActions.prepare,
+            alias: PriceListHandlers.createPriceLists.aliases.priceLists,
+          },
+        },
+        PriceListHandlers.createPriceLists
+      ),
+      compensate: pipe(
+        {
+          invoke: {
+            from: CreatePriceListActions.createPriceList,
+            alias: PriceListHandlers.removePriceLists.aliases.priceLists,
+          },
+        },
+        PriceListHandlers.removePriceLists
+      ),
+    },
+  ],
+])
+
+WorkflowManager.register(Workflows.CreatePriceList, workflowSteps, handlers)
+
+export const createPriceLists = exportWorkflow<
+  WorkflowTypes.PriceListWorkflow.CreatePriceListWorkflowInputDTO,
+  {
+    priceList: PricingTypes.CreatePriceListDTO
+  }[]
+>(
+  Workflows.CreatePriceList,
+  CreatePriceListActions.createPriceList,
+  async (data) => data
```

Comment:
todo-later: The preparation function should also have the context/container. further more, the current exportWorkflow is creating a container if it is not provided but in fact, it is creating an array of loaded modules where it should probably create a new medusa container. lets discuss that @carlos-r-l-rodrigues